core/src/main/scala/kafka/server/metadata/BrokerMetadataSnapshotter.scala

```diff
@@ -59,11 +60,19 @@ class BrokerMetadataSnapshotter(
       val writer = writerBuilder.build(
         image.highestOffsetAndEpoch().offset,
         image.highestOffsetAndEpoch().epoch,
-        lastContainedLogTime
-      )
+        lastContainedLogTime)
       if (writer.nonEmpty) {
         _currentSnapshotOffset = image.highestOffsetAndEpoch().offset
-        info(s"Creating a new snapshot at offset ${_currentSnapshotOffset}...")
+
+        var stringReasons: Set[String] = Set()
+
+        snapshotReasons.foreach(r => stringReasons += r.toString)
+
+        if (stringReasons.isEmpty){
+          stringReasons += SnapshotReason.UnknownReason.toString
+        }
+
+        info(s"Creating a new snapshot at offset ${_currentSnapshotOffset} because, ${stringReasons.mkString(" and ")}")
```

Comment:
We directly use snapshotReasons.mkString(" and ") here and remove `stringReasons` and related code.

Reply:
Made this change